src/pages/RFOX/components/Claim/ClaimConfirm.tsx

```diff
@@ -37,23 +58,157 @@ const CustomRow: React.FC<RowProps> = props => <Row fontSize='sm' fontWeight='me
+  )
+
+  const stakingAssetAccountNumberFilter = useMemo(() => {
+    return {
+      assetId: claimQuote.claimAssetId,
+      accountId: claimQuote.claimAssetAccountId,
+    }
+  }, [claimQuote.claimAssetAccountId, claimQuote.claimAssetId])
+
+  const stakingAssetAccountNumber = useAppSelector(state =>
+    selectAccountNumberByAccountId(state, stakingAssetAccountNumberFilter),
   )
 
+  const claimAmountUserCurrency = useMemo(
+    () =>
+      bnOrZero(stakingAmountCryptoPrecision)
+        .times(claimAssetMarketDataUserCurrency.price)
+        .toFixed(),
+    [claimAssetMarketDataUserCurrency.price, stakingAmountCryptoPrecision],
+  )
+
+  const callData = useMemo(() => {
+    if (!stakingAsset) return
+
+    return encodeFunctionData({
+      abi: foxStakingV1Abi,
+      functionName: 'withdraw',
+      args: [BigInt(claimQuote.index)],
+    })
+  }, [stakingAsset, claimQuote.index])
+
+  const {
+    mutateAsync: handleClaim,
+    isIdle: isClaimMutationIdle,
+    isPending: isClaimMutationPending,
+    isSuccess: isClaimMutationSuccess,
+  } = useMutation({
+    mutationFn: async () => {
+      if (!wallet || stakingAssetAccountNumber === undefined || !stakingAsset || !callData) return
+
+      const adapter = assertGetEvmChainAdapter(stakingAsset.chainId)
+
+      const buildCustomTxInput = await createBuildCustomTxInput({
+        accountNumber: stakingAssetAccountNumber,
+        adapter,
+        data: callData,
+        value: '0',
+        to: RFOX_PROXY_CONTRACT_ADDRESS,
+        wallet,
+      })
+
+      const txId = await buildAndBroadcast({
+        adapter,
+        buildCustomTxInput,
+        receiverAddress: CONTRACT_INTERACTION, // no receiver for this contract call
+      })
+
+      return txId
+    },
+    onSuccess: (txId: string | undefined) => {
+      if (!txId) return
+
+      setClaimTxid(txId)
+    },
+  })
+
+  const isGetClaimFeesEnabled = useMemo(
+    () =>
+      Boolean(
+        isClaimMutationIdle &&
+          stakingAssetAccountNumber !== undefined &&
+          wallet &&
+          stakingAsset &&
+          callData &&
+          feeAsset &&
+          feeAssetMarketDataUserCurrency,
+      ),
+    [
+      isClaimMutationIdle,
+      stakingAssetAccountNumber,
+      wallet,
+      stakingAsset,
+      callData,
+      feeAsset,
+      feeAssetMarketDataUserCurrency,
+    ],
+  )
+
+  const {
+    data: claimFees,
+    isLoading: isClaimFeesLoading,
+    isSuccess: isClaimFeesSuccess,
+  } = useQuery({
+    ...reactQueries.common.evmFees({
+      to: RFOX_PROXY_CONTRACT_ADDRESS,
+      from: stakingAssetAccountAddress,
+      accountNumber: stakingAssetAccountNumber!, // see isGetStakeFeesEnabled
+      data: callData!, // see isGetStakeFeesEnabled
+      value: '0', // contract call
+      wallet: wallet!, // see isGetStakeFeesEnabled
+      feeAsset: feeAsset!, // see isGetStakeFeesEnabled
+      feeAssetMarketData: feeAssetMarketDataUserCurrency!, // see isGetStakeFeesEnabled
```

Comment:
Note to wait for resolution of conversation surrounding this patten here:
https://github.com/shapeshift/web/pull/7020#pullrequestreview-2100537436